src/dsp_tools/xmllib/helpers.py

```diff
@@ -1264,6 +1264,49 @@ def create_list_from_string(string: str, separator: str) -> list[str]:
     return [strpd for x in string.split(separator) if (strpd := x.strip())]
 
 
+def create_list_if_input_is_non_empty(input_value: Any, separator: str) -> list[str]:
+    """
+    If the input value is a non-empty string it returns the separated list.
```

Comment:
Here you directly start with special cases. Instead, I would expect 1 simple introductory sentence which describes the problem that this function solves.

For example: Split up a string (or other input value) into a list of strings, using the provided separator.